App/src/main/java/vut/fit/gja2023/app/service/CourseService.java

```diff
@@ -110,6 +116,58 @@ public void addStudentToCourse(@NotNull CourseBo course, @NotNull UserBo student
             .filter(assignment -> !assignment.isTeam())
             .forEach(assignment -> projectService.createProject(student, assignment));
     }
+    
+    public void upsertTeams(@NotNull CourseBo course, @NotNull Map<String, List<String>> teamsStudents) {
+        
+        teamsStudents.forEach((teamName, loginsList) -> {
+            TeamBo newTeam = teamService.generateNewTeam(teamName, course);
+            
+            List<UserBo> existingStudents = userRepository.findAllByLogins(loginsList);
+            
+            loginsList.forEach(login -> {
+                
+                UserBo existingStudent = existingStudents.stream()
+                    .filter(student -> login.matches(student.getLogin()))
+                    .findFirst()
+                    .orElse(null);
+
+                //student already exists
+                if (existingStudent != null) {
+                    
+                    // student is not yet attending this course
+                    if (!existingStudent.getStudiedCourses().contains(course)) {
+                        addStudentToCourse(course, existingStudent);
+                    }
+                    
+                } else {
```

Comment:
pokud student neexistuje, nemá podle mě smysl ho vytvářet, logicky učitel nejdříc naimporuje do předmětu studenty a až pak teamy, ale serem na to

Reply:
Ale co když ne? Třeba i omylem.